web/src/views/events/EventView.tsx

```diff
@@ -441,10 +452,16 @@ function DetectionReview({
           />
         )}
 
-        {itemsToReview == 0 && (
-          <div className="size-full flex flex-col justify-center items-center">
+        {currentItems == null && (
+          <div className="absolute left-1/2 -translate-x-1/2 top-1/2 -translate-y-1/2">
+            <ActivityIndicator />
+          </div>
+        )}
+
+        {currentItems != null && currentItems.length == 0 && (
```

Comment:
could this just be `{currentItems?.length === 0 && (`

`===` is for a strict equality comparison making sure the length is exactly zero.